src/java.base/share/classes/java/lang/classfile/ClassSignature.java

```diff
@@ -34,6 +35,11 @@
 /**
  * Models the generic signature of a class or interface, as defined by JVMS
  * {@jvms 4.7.9.1}.
+ * <p>
+ * Similar to nominal descriptors from {@link java.lang.constant}, {@code
+ * ClassSignature} can represent generic signatures that cannot be represented in
+ * classfile.  There is no classfile representation checks for string or nominal
+ * descriptor arguments passed to static factory methods in this class.
```

Comment:
I think this information is misleading. I've read it multiple times and still not sure what is the key takeaway for user.